src/cascadia/TerminalCore/Terminal.cpp

```diff
@@ -587,24 +587,53 @@ std::wstring Terminal::GetHyperlinkAtViewportPosition(const til::point viewportP
 
 std::wstring Terminal::GetHyperlinkAtBufferPosition(const til::point bufferPos)
 {
-    auto attr = _activeBuffer().GetCellDataAt(bufferPos)->TextAttr();
+    // Case 1: buffer position has a hyperlink stored in the buffer
+    const auto attr = _activeBuffer().GetCellDataAt(bufferPos)->TextAttr();
     if (attr.IsHyperlink())
     {
-        auto uri = _activeBuffer().GetHyperlinkUriFromId(attr.GetHyperlinkId());
-        return uri;
+        return _activeBuffer().GetHyperlinkUriFromId(attr.GetHyperlinkId());
     }
-    // also look through our known pattern locations in our pattern interval tree
-    auto viewportPos = bufferPos;
-    _GetVisibleViewport().ConvertToOrigin(&viewportPos);
-    const auto result = GetHyperlinkIntervalFromViewportPosition(viewportPos);
+
+    // Case 2: buffer position may point to an auto-detected hyperlink
+    // Case 2 - Step 1: get the auto-detected hyperlink
+    std::optional<interval_tree::Interval<til::point, size_t>> result;
+    const auto visibleViewport = _GetVisibleViewport();
+    if (visibleViewport.IsInBounds(bufferPos))
+    {
+        // Hyperlink is in the current view, so let's just get it
+        auto viewportPos = bufferPos;
+        visibleViewport.ConvertToOrigin(&viewportPos);
+        result = GetHyperlinkIntervalFromViewportPosition(viewportPos);
+        if (result.has_value())
+        {
+            result->start = _ConvertToBufferCell(result->start);
+            result->stop = _ConvertToBufferCell(result->stop);
+        }
+    }
+    else
+    {
+        // Hyperlink is outside of the current view.
+        // We need to find if there's a pattern at that location.
+        const auto patterns = _activeBuffer().GetPatterns(bufferPos.y, bufferPos.y);

```

Comment:
what happens when a hyperlink starts above the screen but wraps onto another line? Will we accidentally capture only half of it?

Reply:
Welp. Found this out during a bug bash and now it's a line item in #13854. See https://github.com/microsoft/terminal/issues/13854#issuecomment-1233590998 for details on what's going on.